main/src/cgeo/geocaching/ui/CacheListAdapter.java

```diff
@@ -180,8 +183,11 @@ public CacheComparator getCacheComparator() {
         if (cacheComparator == null && series) {
             return NameComparator.INSTANCE;
         }
+        if (cacheComparator == null && targetCoords != null) {
+            return new TargetDistanceComparator(targetCoords);
+        }
         if (cacheComparator == null) {
-            return DistanceComparator.DISTANCE_TO_GLOBAL_GPS;
+            return DistanceComparator.INSTANCE;
```

Comment:
INSTANCE is the standard name used for singleton pattern, it is known to be used for classes where only one instance exists and is used.

 If class `DistanceComparator` is meant to be a Singleton class which is solely used for compare to global GPS then I suggest renaming it to e.g. `GlobalGPSDistanceConparator`

If class `DistanceComparator` should also be used for other cases (like it was implemented before) then I would strongly suggest leaving the constant name as it was - the class is not a Singleton in this case.

Reply:
> If class `DistanceComparator` is meant to be a Singleton class which is solely used for compare to global GPS then I suggest renaming it to e.g. `GlobalGPSDistanceConparator`

Yes, it is designed as singleton class, as userGPS will be the only value which can change. For other use cases we have `AbstractDistanceComparator` and `TargetDistanceComparator`. The comparator provider is BTW designed that way that every comparator requires it's own class, so every comparator class needs to be a singleton. I will rename the class accordingly.

Reply:
My first implementation indeed was to create simply a new instance of DistanceComparator (without new classes), but it was not working without deeper refactoring...